### ISSUE TITLE
[BUG][iPadOS] Accessibility Bug - Alt text for image description not visible and cannot be entered

### ISSUE THREAD

**Issue by wayneca** [mastodon/mastodon-ios#890]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Using iOS Mastodon app on an iPad, enter an image, the entire image usually cannot be seen.   A string overlaying the bottom of the image must be selected so that text can be entered to describe the image for accessibility.   However, the string is often not visible because not all of the image can be seen.  If you scroll to the bottom to see the string, when you release your finger, the image bounces back,  so the bottom of the image is not seen again. (Added later: this happens if the virtual keyboard is large size, even if not shown because physical keyboard is present)

There is no way  to scroll to see the alt text overlay and select it to give it focus and to type at the same time.   Being able to enter alt text  is essential for providing auditory content used by people with low vision.  

Here is a recent discussion of this bug on a leading Mastodon art source 
.  https://fosstodon.org/@photonstopper@mastodon.art/109699888660622737 

There are other issues describing this bug, but they are not listed as bugs and have not been assigned to someone in the project.  #864 and #813 Marking one of those as a bug and assigning it would be enough to close this.  I'm concerned that it isn't getting project attention because it may not be clear enough that this is a serious accessibility bug.

Workaround: if you add an additional image at the bottom that you will delete before posting, then you can see and edit the overlay for adding alt text to the images above that last image.  then you can delete the last dummy image.  Or make the virtual keyboard small (but the small keyboard is a problem for people with limited vision).

### Expected Behavior

Expect to be able to see the field in which to enter the alt text description and to be able to type into it.  This works on the "official" Android Mastodon app and on the Web app, but not on the iPadOS app.

### Steps To Reproduce

1. On an iPad, insert an image in a toot
2. Observe that you cannot see the input line at the bottom of the image, because it is below the part of the image displayed.  
3. Scroll using touch to see the bottom of the image so you can see the text suggesting to enter a description
4. Observe that when you stop scrolling and lift finger, it scrolls back so can't see where the "alt" text must be entered, so cannot enter the "alt" text for accessibility.

### Environment

```markdown
- Device: iPadAir 4th generation
- OS: iPadOS
- Version: 16.2
- Build: 20C65
```

### Anything else?

No

**Comment by wayneca** [comment]
I updated the description above when I realized the problem is caused by scrolling bouncing back a little when released so that you can never scroll to the bottom of the image because it bounces back when released to cover just where the alt text entry overlay is.  (You also cannot set the focus point in the iOS app.  If you include multiple images, it is only the bottom one that you cannot scroll to and add alt text. This is relevant to #864nas well

**Comment by j-f1** [comment]
I’m unable to reproduce this on an iPhone, but I tested on a simulated iPad, and I think I’ve identified a bug in the code that computes the keyboard size to enable the toolbar to avoid it. PR coming soon, which should allow you to scroll to the bottom image.

**Comment by wayneca** [comment]
On my iPad, this bug happens whether the physical keyboard is attached or not.  So it happens even when there is no virtual keyboard on the screen at all and lots of room.  It seems to try to avoid the large keyboard even if it isn't there.  In any case, it  still shouldn't refuse to stay at  the bottom of the image when scrolled. 

 If I insert 2 images, I can scroll to the alt text on the first one and enter it, but I can't scroll to the  image at the bottom.  It doesn't want that alt text in the viewport at the bottom.  

I jinstalled on my iphone and  confirm that the bug does not happen on my iphone.  It does happen on the iPad.

**Comment by wayneca** [comment]
On an iPad, if you shrink the keyboard to half size, it moves to the bottom right and the buttons (for images, emojis, etc) move down to the left of the keyboard.  Then the image can be scrolled to the bottom and you can give focus to and type into the overlay on the image for alt text.  Mentioning this in case it helps to see how it miscalculates.

**Comment by wayneca** [comment]
The same bug causes a different problem in bug #946
